lib/mergeusersearch.php

```diff
@@ -60,10 +60,16 @@ public function search_users($input, $searchfield){
         switch($searchfield){
             case 'id': // search on id field
 
-                $params = array(
-                    'userid' => $input,
-                );
-                $sql = 'SELECT * FROM {user} WHERE id = :userid';
```

Comment:
Thanks @leonstr for this patch. But, unfortunatelly, this is not the way of doing. By intheritance of the beginning of the development of this plugin, there are some places with specific code and conditions according to types of database. 

However, the moodle_database class affords this specificities very well. What we need to call, however, are the proper API to the `$DB` instance, like the `$DB->get_in_or_equal(...)` that already catches these particularities of null values and other specific dababase API.

I mean, the error you are providing the patch is that we originally did not considered using the `$DB` API properly. 

Could you, please, use the `$DB->get_in_or_equal()` API and remake this part?

Reply:
In addition, this DB API would also consider edge cases, like empty strings, null values and so, producing the necessary comparison condition for the WHERE clause. Currently the ctype_digit() call matches both for empty string and null values. Empty string would be a false positive.

Reply:
I investigated `$DB->get_in_or_equal()` but I don't think this handles the issue matching strings with integer columns in PostgreSQL. In fact, this only does any special handling for this database type if there are >= 65535 elements in the `IN (...)`.

I have reworked the use of `ctype_digit()` to `is_string($input) && !ctype_digit($input)` which should now correctly handle, e.g., "", "123", 123, and null.

I've also reworked `MergeUserSearch->search_users()` to use `$DB->get_records_select()` instead of `$DB->get_records_sql().